service/matching/matching_engine.go

```diff
@@ -1827,6 +1867,28 @@ func (e *matchingEngineImpl) ListNexusEndpoints(ctx context.Context, request *ma
 	}
 }
 
+func (e *matchingEngineImpl) checkNexusEndpointsOwnership() (bool, error) {
+	self := e.hostInfoProvider.HostInfo().Identity()
+	owner, err := e.serviceResolver.Lookup(nexusEndpointsTablePartitionRoutingKey)
+	if err != nil {
+		return false, fmt.Errorf("cannot resolve Nexus endpoints partition owner: %w", err)
+	}
+	return owner.Identity() == self, nil
+}
+
+func (e *matchingEngineImpl) notifyIfNexusEndpointsOwnershipLost() {
+	isOwner, err := e.checkNexusEndpointsOwnership()
+	if err != nil {
+		e.logger.Error("Failed to check Nexus endpoints ownerhip", tag.Error(err))
+		return
+	}
+	if !isOwner {
+		ch := e.nexusEndpointsOwnershipLostCh
+		e.nexusEndpointsOwnershipLostCh = make(chan struct{})
```

Comment:
is there a race here? the List call wait on this channel right after this line?

Reply:
Hmmm... yeah, let me change to first close and then replace.

Reply:
Actually it doesn't help much, there can still be a race from when we check if we're the owner to when we start listening on this channel.

Reply:
I fixed, let me just verify it and push.